contracts/Exchanger.sol

```diff
@@ -598,6 +634,36 @@ contract Exchanger is Owned, MixinSystemSettings, IExchanger {
         );
     }
 
+    function _exchangeAtomically(
+        address from,
+        bytes32 sourceCurrencyKey,
+        uint sourceAmount,
+        bytes32 destinationCurrencyKey,
+        address destinationAddress,
+    )
+        internal
+        returns (
+            uint amountReceived,
+            uint fee
+        )
+    {
+        // TODO:
+        //   1. ensure can exchange (also checks if any chainlink flags are up)
+        //     - check here if synths are allowed to be atomically exchanged or leave that up to ExchangeRates?
+        //   2. settle any previous exchanges
+        //   3. convert source synth into sUSD and update volume counter
+        //   4. get amounts and rates (_getAmountsForAtomicExchangeMinusFees())
+        //   5. check _isSynthRateInvalid()
+        //     - if we also have the chainlink price at this stage, we can check the circuit breaker with it
+        //   6. burn/issue via _convert()
+        //   7. remit sUSD fee via chainlink price
+        //
+        //   - avoid persisting exchange via appendExchange() to avoid fee reclamation state
+        //   - do we need to update the debt cache (_updateSNXIssuedDebtOnExchange())? If so, we
+        //     should use the chainlink rates?
+        //   - emitSynthExchange() is ok or should another event be used?
```

Comment:
Main meat of the exchange functionality, might be good to sanity check these steps.